lib/mayaUsd/resources/ae/usdschemabase/ae_template.py

```diff
@@ -492,6 +494,35 @@ def createAppliedSchemasSection(self):
                     typeName = self.sectionNameFromSchema(typeName)
                     self.createSection(typeName, attrs, False)
 
+    @staticmethod
+    def getAETemplateForCustomCallback():
+        global _aeTemplate
+        return _aeTemplate
+
+    def createCustomCallbackSection(self):
+        '''Special custom callback section that gives users the opportunity to add
+        layout section(s) to the AE template.
+        See https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/resources/ae/usdschemabase/Attribute-Editor-Template-Doc.md
+        for more info.
+        '''
+
+        # Create the callback context/data (empty).
+        cbContext = {
+            'ufe_path_string' : ufe.PathString.string(self.item.path())
+        }
+        cbContextDict = Vt._ReturnDictionary(cbContext)
+        cbDataDict = Vt._ReturnDictionary({})
+
+        # Trigger the callback which will give other plugins the opportunity
+        # to add controls to our AE template.
+        global _aeTemplate
+        try:
+            _aeTemplate = self
```

Comment:
Set our template into the global variable so a callback can retrieve it using the new function `getAETemplateForCustomCallback` above. They can use this to add/suppress controls (which then follows all the rules like show array attributes, nice name, etc).

Reply:
There was not any way to pass it as a parameter to the callback instead? I find that having to go through a global not great, but if there are limitation onbacks, then I suppose we have no choice.

Reply:
No, not using the existing UICallback class which takes VtDictionary. I talked to PierreT about this and confirmed using the variable is okay. Its already done in our Ufe support plugin (in Maya) for the command.